plugins/woocommerce-admin/client/marketing/data-multichannel/reducer.ts

```diff
@@ -66,21 +64,23 @@ export const reducer: Reducer< State, Action > = (
 			};
 
 		case TYPES.RECEIVE_CAMPAIGNS:
+			const { meta } = action;
+			const key = `${ meta.page }-${ meta.perPage }`;
```

Comment:
> This is kinda personal opinion: I would prefer not to use `-` as a data separator because `-` is often used in slugs and in URLs, and may cause confusion. I would prefer something else like `|` as data separator in a string.

Globally searching the codebase of this repo with ``/const \w+ = (['"`]).+?\|.+?\1/i`` and ``/\+ (['"`])\|\1 \+/`` regex patterns only got two instances:

1. https://github.com/woocommerce/woocommerce/blob/6b77f71ea7312a18ac35047cce38ffa5574eece4/.github/actions/tests/slack-summary-on-release/slack-blocks/scripts/create-result-block/utils/set-element-text.js#L5
2. https://github.com/woocommerce/woocommerce/blob/6b77f71ea7312a18ac35047cce38ffa5574eece4/plugins/woocommerce/client/legacy/js/admin/api-keys.js#L88

Searching with ``const \w*(id|code|key) = (['"`]).+?-.+?\2`` and ``/\+ (['"`])-\1 \+/`` got dozens of instances:

![image](https://github.com/woocommerce/woocommerce/assets/17420811/40a83b78-2830-4448-bd8c-c1971e9eb65d)

It's certainly commonly used for slugs or URLs, and quite often for IDs or keys as well. Also, I don't get the point of reading ``const key = `${ meta.page }-${ meta.perPage }`;`` would be confused with a slug or a URL.

If a separator does not appear in the value range of `page` and `perPage`, is not an unrecognizable character, and does not violate convention, then I think it's reasonable.

> Also, I have a thought to make the key more readable, e.g. "page:1|perPage:10". I think that is helpful when we look at redux dev tools and when troubleshooting. But on the other hand, this may also be considered too verbose.

Since there are only two parameters, I don't think it's complicated enough to require a highly readable key.

Reply:
> It's certainly commonly used for slugs or URLs, and quite often for IDs or keys as well. Also, I don't get the point of reading `` const key = `${ meta.page }-${ meta.perPage }`; `` would be confused with a slug or a URL.

I guess I should have illustrated it with a scenario. 😄 

Imagine an example scenario we see a string "hello-world-1-2", in a URL path, query string, key ID etc. At first glance, we may think it is _one data piece_ of string. However, if we dig into the code, we may find that it is actually constructed from _multiple pieces of data_, for example `${ stringHelloWorld1 }-${ number2}`. If we use another character to construct the string like `${ stringHelloWorld1 }|${ number2}`, it would produce "hello-world-1|2", which would indicate that it is more likely to be constructed from multiple data pieces.

In summary, I guess what I want to convey is that: because "-" commonly appears _within one string data piece_, I would prefer not to use it as a data separator to join _multiple data pieces_. If we do that, it makes it a bit more difficult to tell from the final string whether it comes from one string data piece or it is generated from multiple data pieces. And it may be difficult to deconstruct the final string back into the individual data pieces.

> If a separator does not appear in the value range of `page` and `perPage`, is not an unrecognizable character, and does not violate convention, then I think it's reasonable.

Yes, for our case here, I think it is okay and reasonable. I was just thinking about more complicated scenarios, and we may want to be consistent across scenarios.

We can stick with what we have here. 🙂 👌

Reply:
Thanks for the explanation!

> Imagine an example scenario we see a string "hello-world-1-2", in a URL path, query string, key ID etc. At first glance, we may think it is _one data piece_ of string. However, if we dig into the code, we may find that it is actually constructed from _multiple pieces of data_, for example `${ stringHelloWorld1 }-${ number2}`.

I would say this scenario is not analogous to the use case here. If "-" is possible in the value range of variables to be concatenated, then it should not be used as a concatenation character in the first place. Moreover, perhaps the premise that it is considered as a "data (piece)" to be discussed is not quite accurate either.

The intent of this code should be clear: it wants to generate a non-colliding data key to be concatenated only and doesn't need to be separated back for somewhat uses.

> In summary, I guess what I want to convey is that: because "-" commonly appears _within one string data piece_, I would prefer not to use it as a data separator to join _multiple data pieces_. If we do that, it makes it a bit more difficult to tell from the final string whether it comes from one string data piece or it is generated from multiple data pieces.

I would suggest not just looking at the final string to determine/infer how it came to be, instead, exactly find where it came from. It's like saying that when seeing a string "1|5", a dev still has to find out where it came from in order to truly determine how it came to be, and it doesn't make the whole thing any easier or harder.

> And it may be difficult to deconstruct the final string back into the individual data pieces.

If the character meets the principles mentioned at the end of [this comment](https://github.com/woocommerce/woocommerce/pull/41211#discussion_r1411813274), I believe it's not that difficult to separate it.

> I was just thinking about more complicated scenarios, and we may want to be consistent across scenarios.

To discuss consistency across (more complicated) scenarios, I guess [Discussions](https://github.com/woocommerce/woocommerce/discussions) would be a better place.